test_scripts/MobileProjection/Defects/3142_01_Absence_of_misplaced_OnVideoDataStreaming.lua

```diff
@@ -0,0 +1,84 @@
+--   - not unregister App_1 since timeout is not yet expired
+-- 8. App_1 stops streaming within 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - not send Navi.OnVideoDataStreaming(false) notification to HMI since App_2 still continue streaming
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("test_scripts/MobileProjection/Defects/common")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Apps Configuration ]]
+common.app.getParams(1).appHMIType = { "PROJECTION" }
+common.app.getParams(2).appHMIType = { "NAVIGATION" }
+
+--[[ Local Functions ]]
+local function stopStreaming(pAppId, pServiceId)
+  common.hmi.getConnection():ExpectRequest("Navigation.StopStream", { appID = common.app.getHMIId(pAppId)})
+  :Do(function(_, data) common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { }) end)
+  common.mobile.getSession(pAppId):ExpectEndService(pServiceId)
+  :Do(function() common.mobile.getSession(pAppId):SendEndServiceAck(pServiceId) end)
+end
+
+local function startStreaming(pAppId, pServiceId)
+  common.mobile.getSession(pAppId):StartService(pServiceId)
+  :Do(function()
+      common.mobile.getSession(pAppId):StartStreaming(pServiceId, common.streamFiles[pAppId], 160*1024)
+      common.log("App " .. pAppId .." starts streaming ...")
+      common.streamingStatus[pAppId] = true
+    end)
+  common.hmi.getConnection():ExpectRequest("Navigation.StartStream", { appID = common.app.getHMIId(pAppId) })
+  :Do(function(_, data) common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { }) end)
+end
+
+local function activateApp2()
+  local cid = common.hmi.getConnection():SendRequest("SDL.ActivateApp", { appID = common.app.getHMIId(2) })
+  common.hmi.getConnection():ExpectResponse(cid)
+  common.mobile.getSession(2):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", videoStreamingState = "STREAMABLE" })
+  :Do(function()
+      startStreaming(2, 11)
+    end)
+  stopStreaming(1, 11)
+  common.hmi.getConnection():ExpectNotification("Navigation.OnVideoDataStreaming",
+    { available = false }, { available = true }):Times(2)
+end
+
+local function stopStreamingWONotification()
+  common.stopStreaming(1)
+  common.hmi.getConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = false }):Times(0)
+  common.wait(10000)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions, {{ StopStreamingTimeout = 3000, VideoDataStoppedTimeout = 3000 }})
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App 1", common.registerApp, { 1 })
+runner.Step("Register App 2", common.registerApp, { 2 })
+runner.Step("Activate App 1", common.activateApp, { 1 })
+
+runner.Title("Test")
```

Comment:
Please add check for Audio streaming.

Reply:
@GetmanetsIrina Done in e8638e6d34fdaa7c5d5fb714b51f0b69ce56c690